bw2io/backup.py

```diff
@@ -55,40 +62,155 @@ def backup_data_directory(
+        The new data directory path.
+
+    Raises
+    ------
+    ValueError:
+        If the file doesn't exist
+    PermissionError:
+        If the data directory is not writable.
+    Exception:
+        If there's an attempted path traversal in the tar file.
+
+    Returns
+    -------
+    data_dir (Path):
+        The path of the restored data directory.
+
+    See Also
+    --------
+    bw2io.backup.backup_data_directory: To backup the data directory.
+    """
+
+    # check if file exists
+    fp = Path(fp)
+    if not fp.is_file():
+        raise ValueError(f"Can't find file at path: {fp}")
+
+    # if new_data_dir, set the environment variable (this will not be persistent, I only know how to do that on Linux or in a venv...)
+    if new_data_dir:
+        os.environ["BRIGHTWAY2_DIR"] = str(new_data_dir)
+        data_dir = Path(new_data_dir)
+    else:
+        data_dir = Path(projects._base_data_dir)
+
+    # Confirm that the user wants to overwrite the data directory, if it exists
+    if data_dir.is_dir():
+        confirm_overwrite = input(
+            f"This will overwrite your existing brightway data directory '{data_dir}'.\nAre you really, really sure that you want to do that? (y/n) "
+        )
+        if confirm_overwrite.lower() == "y":
+            shutil.rmtree(data_dir)
+        else:
+            print("Aborting...")
+            return
+
+    # Extract the tar file
+    print(
+        "Restoring brightway data directory from backup archive - this could take a few minutes..."
+    )
+    with tempfile.TemporaryDirectory() as td:
+        with tarfile.open(fp, "r:gz") as tar:
+
+            def is_within_directory(directory, target):
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+
+                return prefix == abs_directory
+
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception("Attempted Path Traversal in Tar File")
+
+                tar.extractall(path, members, numeric_owner=numeric_owner)
+
+            safe_extract(tar, td)
+
+        # Find single extracted directory; don't know it ahead of time
+        extracted_dir = [
+            (Path(td) / dirname)
+            for dirname in Path(td).iterdir()
+            if (Path(td) / dirname).is_dir()
+        ]
+        extracted_path = extracted_dir[0]
+        shutil.copytree(extracted_path, data_dir)
+
+    print(f"Restored brightway data to directory: {data_dir}")
+
+    # this block is maybe totally the wrong way to do it --> could be done better or removed
+    if data_dir != Path(projects._base_data_dir):
+        projects.change_base_directories(data_dir)
```

Comment:
Probably better to give a reference to https://docs.brightway.dev/en/latest/content/faq/data_management.html#how-do-i-change-my-data-directory